faq.md

```diff
@@ -14,6 +14,84 @@ Here you can find a few common questions regarding the framework.
 1. TOC
 {:toc}
 
+## Blocking method calls and endless loops
+
+If you perform any blocking calls (i.e. `Thread.sleep`, endless loops or `I/O`), you should move them to your main 
+method. Otherwise, it could block the communication and initialization of the multitier object.
+
+```scala
+  // Avoid:
+  on[Server] {
+    while (running) {
+        ...
+    }
+  }
+
+  // Better:
+  def main() = on[Server] {
+    // perform it here   
+  }
+```
+
+## Defining main method in multiple peers
+
+If your application requires a method with the same name, but with different semantics on different peers, you have
+to define a common peer type and specify the return type explicitly. You can then combine the logic using the `and`
+keyword between two peers. If you have more than two peers, you need to add correct parenthesis like below. So
+the implementation of peer `Client` is combined with the tuple of `(Controller and Server)`.
+
+```scala
+  def main(): Unit on Peer = on[Client] {
```

Comment:
add the `@peer type` definitions of `Client`, `Controller` and `Server` to the snippet to show the "common peer" super type relation.